python/cuml/test/test_metrics.py

```diff
@@ -160,6 +163,186 @@ def test_rand_index_score(name, nrows):
+    ref = sk_mi_score(a_dev, b_dev)
+    np.testing.assert_almost_equal(score, ref, decimal=4)
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_mutual_info_score(use_handle):
+    def assert_ours_equal_sklearn(ground_truth, predictions):
+        a = np.array(ground_truth, dtype=np.int32)
+        b = np.array(predictions, dtype=np.int32)
+        assert_mi_equal_sklearn(a, b, use_handle=use_handle)
+
+    assert_ours_equal_sklearn([0, 0, 1, 1], [1, 1, 0, 0])
+    assert_ours_equal_sklearn([0, 0, 1, 1], [0, 0, 1, 1])
+    assert_ours_equal_sklearn([0, 0, 1, 1], [0, 0, 1, 2])
+    assert_ours_equal_sklearn([0, 0, 1, 1], [0, 1, 2, 3])
+    assert_ours_equal_sklearn([0, 0, 1, 1], [0, 1, 0, 1])
+    assert_ours_equal_sklearn([0, 0, 1, 1], [0, 0, 0, 0])
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_mutual_info_score_big_array(use_handle):
+    def assert_equal_sklearn(random_generation_lambda):
+        rng = np.random.RandomState(1234)  # makes it reproducible
+        a = random_generation_lambda(rng)
+        b = random_generation_lambda(rng)
+        assert_mi_equal_sklearn(a, b, use_handle=use_handle)
+
+    assert_equal_sklearn(lambda rng: rng.randint(0, 1000, int(10e4),
+                                                 dtype=np.int32))
+    assert_equal_sklearn(lambda rng: rng.randint(-1000, 1000, int(10e4),
+                                                 dtype=np.int32))
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_homogeneity_completeness_symmetry(use_handle):
+    def assert_hom_com_sym(random_generation_lambda, seed=1234):
+        a_dev, b_dev = generate_random_labels(random_generation_lambda)
+        handle, stream = get_handle(use_handle)
+        hom = cuml.metrics.homogeneity_score(a_dev, b_dev, handle=handle)
+        com = cuml.metrics.completeness_score(a_dev, b_dev, handle=handle)
+        np.testing.assert_almost_equal(hom, com, decimal=3)
+
+    assert_hom_com_sym(lambda rng: rng.randint(0, 2, int(10e3),
+                                               dtype=np.int32))
+    assert_hom_com_sym(lambda rng: rng.randint(-5, 20, int(10e3),
+                                               dtype=np.int32))
+    assert_hom_com_sym(lambda rng:
+                       rng.randint(int(-10e3), int(10e3), int(10e3),
+                                   dtype=np.int32))
+
+
+@pytest.mark.parametrize('use_handle', [True, False])
+def test_completeness_score(use_handle):
+    def score_labeling(ground_truth, predictions):
+        a = np.array(ground_truth, dtype=np.int32)
+        b = np.array(predictions, dtype=np.int32)
+
+        a_dev = cuda.to_device(a)
+        b_dev = cuda.to_device(b)
+
+        handle, stream = get_handle(use_handle)
+
+        return cuml.metrics.completeness_score(a_dev, b_dev, handle=handle)
+
+    # Perfect labelings are complete
+    np.testing.assert_almost_equal(score_labeling([0, 0, 1, 1], [1, 1, 0, 0]),
```

Comment:
As in the other view, I'd suggest doing this with parametrize instead.

Reply:
Agreed.